services/slack_grid/parse.go

```diff
@@ -0,0 +1,383 @@
+package slack_grid
+
+import (
+	"archive/zip"
+	"encoding/json"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	slack "github.com/mattermost/mmetl/services/slack"
+	"github.com/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type ChannelFiles string
+
+const (
+	ChannelFilePublic  ChannelFiles = "channels"
+	ChannelFilePrivate ChannelFiles = "groups"
+	ChannelFileDM      ChannelFiles = "dms"
+	ChannelFileGM      ChannelFiles = "mpims"
+)
+
+const ErrFindingTeamID = "Could not find team ID for channel %v, ID: %v"
+const ErrFindingTeamName = "Could not find team name for channel %v"
+
+type BulkSlackExport struct {
+	Public  []slack.SlackChannel
+	Private []slack.SlackChannel
+	GMs     []slack.SlackChannel
+	DMs     []slack.SlackChannel
+}
+
+type Post struct {
+	*slack.SlackPost
+	Team       string `json:"team"`
+	UserTeam   string `json:"user_team"`
+	SourceTeam string `json:"source_team"`
+}
+
+type Channel struct {
+	*slack.SlackChannel
+	Team string `json:"team"`
+}
+
+type BulkTransformer struct {
+	slack.Transformer
+	Teams map[string]string
+
+	// the root path to export slack to for parsing.
+	// defaults to ./tmp/slack_grid
+	dirPath string
+	pwd     string
+}
```

Comment:
Maybe this can be called `GridTransformer` or something similar since it's related specifically to Slack Grid?